components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1485,22 +1484,75 @@ private boolean isNotGroupFilteringSupported(String filterOperation) {
         return userNames;
     }
 
-    private String[] getRoleNames(String filterOperation, String attributeValue) throws org.wso2.carbon.user.core
-            .UserStoreException {
+    private String getSearchAttribute(String filterOperation, String attributeValue, String delimiter) {
 
         String searchAttribute = null;
         if (filterOperation.equalsIgnoreCase(SCIMCommonConstants.CO)) {
-            searchAttribute = FILTERING_DELIMITER + attributeValue + FILTERING_DELIMITER;
+            searchAttribute = delimiter + attributeValue + delimiter;
         } else if (filterOperation.equalsIgnoreCase(SCIMCommonConstants.SW)) {
-            searchAttribute = attributeValue + FILTERING_DELIMITER;
+            searchAttribute = attributeValue + delimiter;
         } else if (filterOperation.equalsIgnoreCase(SCIMCommonConstants.EW)) {
-            searchAttribute = FILTERING_DELIMITER + attributeValue;
+            searchAttribute = delimiter + attributeValue;
+        } else if (filterOperation.equalsIgnoreCase(SCIMCommonConstants.EQ)) {
+            searchAttribute = attributeValue;
         }
+        return searchAttribute;
+    }
 
+    private String[] getRoleNames(String filterOperation, String attributeValue) throws org.wso2.carbon.user.core
+            .UserStoreException {
+
+        String searchAttribute = getSearchAttribute(filterOperation, attributeValue, FILTERING_DELIMITER);
         return ((AbstractUserStoreManager) carbonUM).getRoleNames(searchAttribute, MAX_ITEM_LIMIT_UNLIMITED, true,
                 true, true);
     }
 
+    private String[] getUserNames(String attributeName, String filterOperation, String attributeValue)
+            throws org.wso2.carbon.user.core.UserStoreException {
+
+        String searchAttribute = getSearchAttribute(filterOperation, attributeValue, FILTERING_DELIMITER);
+        return carbonUM.getUserList(attributeName, searchAttribute, UserCoreConstants.DEFAULT_PROFILE);
+    }
+
+    private String[] getGroupList(String attributeName, String filterOperation, String attributeValue)
+            throws org.wso2.carbon.user.core.UserStoreException, IdentitySCIMException {
+
+        String[] userRoleList;
+        if (attributeName.equals(SCIMConstants.GroupSchemaConstants.DISPLAY_URI)
+                || attributeName.equals(SCIMConstants.GroupSchemaConstants.VALUE_URI)) {
+            String[] userList;
+            if (attributeName.equals(SCIMConstants.GroupSchemaConstants.DISPLAY_URI)) {
+                userList = getUserNames(SCIMConstants.UserSchemaConstants.USER_NAME_URI, filterOperation, attributeValue);
```

Comment:
here the attributeName equals to SCIMConstants.GroupSchemaConstants.DISPLAY_URI and the attributeValue should be the corresponding value of the display claim as I assumed, in that case how do you retrieve user names by using SCIMConstants.UserSchemaConstants.USER_NAME_URI since the attributeValue is display claim value?